src/Catalyst.Core.Modules.Dfs.Tests/UnitTests/CoreApi/PubSubApiTest.cs

```diff
@@ -0,0 +1,209 @@
+            await ipfs.StartAsync();
+            try
+            {
+                await ipfs.PubSub.SubscribeAsync(topic, msg => { }, cs.Token);
+                var topics = ipfs.PubSub.SubscribedTopicsAsync().Result.ToArray();
+                Assert.IsTrue(topics.Length > 0);
+                CollectionAssert.Contains(topics, topic);
+            }
+            finally
+            {
+                await ipfs.StopAsync();
+                cs.Cancel();
+            }
+        }
+
+        private volatile int messageCount = 0;
+
+        [Fact]
+        public async Task Subscribe()
+        {
+            messageCount = 0;
+            var ipfs = TestFixture.Ipfs;
+            var topic = Guid.NewGuid().ToString();
+            var cs = new CancellationTokenSource();
+            await ipfs.StartAsync();
+            try
+            {
+                await ipfs.PubSub.SubscribeAsync(topic, msg => { Interlocked.Increment(ref messageCount); }, cs.Token);
+                await ipfs.PubSub.PublishAsync(topic, "hello world!");
+
+                await Task.Delay(100);
+                Assert.AreEqual(1, messageCount);
+            }
+            finally
+            {
+                await ipfs.StopAsync();
+                cs.Cancel();
+            }
+        }
+
+        [Fact]
+        public async Task Subscribe_Mutiple_Messages()
+        {
+            messageCount = 0;
+            var messages = "hello world this is pubsub".Split();
+            var ipfs = TestFixture.Ipfs;
+            var topic = Guid.NewGuid().ToString();
+            var cs = new CancellationTokenSource();
+            await ipfs.StartAsync();
+            try
+            {
+                await ipfs.PubSub.SubscribeAsync(topic, msg => { Interlocked.Increment(ref messageCount); }, cs.Token);
+                foreach (var msg in messages) await ipfs.PubSub.PublishAsync(topic, msg);
+
+                await Task.Delay(100);
+                Assert.AreEqual(messages.Length, messageCount);
+            }
+            finally
+            {
+                await ipfs.StopAsync();
+                cs.Cancel();
+            }
+        }
+
+        [Fact]
+        public async Task Multiple_Subscribe_Mutiple_Messages()
+        {
+            messageCount = 0;
+            var messages = "hello world this is pubsub".Split();
+            var ipfs = TestFixture.Ipfs;
+            var topic = Guid.NewGuid().ToString();
+            var cs = new CancellationTokenSource();
+
+            Action<IPublishedMessage> processMessage = (msg) => { Interlocked.Increment(ref messageCount); };
+            await ipfs.StartAsync();
+            try
+            {
+                await ipfs.PubSub.SubscribeAsync(topic, processMessage, cs.Token);
+                await ipfs.PubSub.SubscribeAsync(topic, processMessage, cs.Token);
+                foreach (var msg in messages) await ipfs.PubSub.PublishAsync(topic, msg);
+
+                await Task.Delay(100);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)